addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-        # is only an indication and not some thing legal.
-        return
-
-    # -------------------------------------------------------------------------
-    # BUSINESS METHODS
-    # -------------------------------------------------------------------------
-
-    def open_cashbox_id(self):
-        self.ensure_one()
-        context = dict(self.env.context or {})
-        if context.get('balance'):
-            context['statement_id'] = self.id
-            if context['balance'] == 'start':
-                cashbox_id = self.cashbox_start_id.id
-            elif context['balance'] == 'close':
-                cashbox_id = self.cashbox_end_id.id
-            else:
-                cashbox_id = False
-
-            action = {
-                'name': _('Cash Control'),
-                'view_mode': 'form',
-                'res_model': 'account.bank.statement.cashbox',
-                'view_id': self.env.ref('account.view_account_bnk_stmt_cashbox_footer').id,
-                'type': 'ir.actions.act_window',
-                'res_id': cashbox_id,
-                'context': context,
-                'target': 'new'
-            }
-
-            return action
-
-    def button_post(self):
-        ''' Move the bank statements from 'draft' to 'posted'. '''
-        if any(statement.state != 'open' for statement in self):
-            raise UserError(_("Only new statements can be posted."))
-
-        self._check_cash_balance_end_real_same_as_computed()
-
-        for statement in self:
-            if not statement.name:
-                statement._set_next_sequence()
-
-        self.write({'state': 'posted'})
+        Currently it removes the statements with no lines, can be extended if needed
+        """
+        self.search([
+            ('line_ids', '=', False),
+            ('write_date', '<', fields.Date.context_today(self))
+        ]).unlink()
+
+    def _post(self):
+        # if any(stmt.state != 'complete' for stmt in self):
+        #     raise UserError(_("You can only validate statements in the complete state."))
+        self.state = 'posted'
         lines_of_moves_to_post = self.line_ids.filtered(lambda line: line.move_id.state != 'posted')
         if lines_of_moves_to_post:
             lines_of_moves_to_post.move_id._post(soft=False)
 
-    def button_validate(self):
-        if any(statement.state != 'posted' or not statement.all_lines_reconciled for statement in self):
-            raise UserError(_('All the account entries lines must be processed in order to validate the statement.'))
-
+        self.line_ids.is_anchor = False
         for statement in self:
-
-            # Chatter.
-            statement.message_post(body=_('Statement %s confirmed.', statement.name))
-
+            statement.line_ids.sorted()[:1].write({'is_anchor': True,
+                                                   'anchor_value': statement.balance_end_real
+                                                   })
```

Comment:
```suggestion
            statement.line_ids.sorted()[:1].write({
                'is_anchor': True,
                'anchor_value': statement.balance_end_real,
           })
```

What if you change the amount of an anchor?

Reply:
amount is not important, we add statements without even showing the affected lines to the user, so the users have to judge based on the balance_end value